pkg/utils/common_test.go

```diff
@@ -87,3 +95,85 @@ func TestGenerateKey(t *testing.T) {
 		})
 	}
 }
+
+func setupMultiConcurrentContexts(t *testing.T) func() {
+	tkgConfigFile, err := os.CreateTemp("", "config")
+	assert.NoError(t, err)
+	err = copy.Copy(filepath.Join("..", "fakes", "config", "tanzu_config.yaml"), tkgConfigFile.Name())
+	assert.NoError(t, err, "Error while copying tanzu config file for testing")
+	os.Setenv("TANZU_CONFIG", tkgConfigFile.Name())
+
+	tkgConfigFileNG, err := os.CreateTemp("", "config_ng")
+	assert.NoError(t, err)
+	os.Setenv("TANZU_CONFIG_NEXT_GEN", tkgConfigFileNG.Name())
+	err = copy.Copy(filepath.Join("..", "fakes", "config", "tanzu_config_ng_2.yaml"), tkgConfigFileNG.Name())
+	assert.NoError(t, err, "Error while copying tanzu-ng config file for testing")
+
+	cleanup := func() {
+		err = os.Remove(tkgConfigFile.Name())
+		assert.NoError(t, err)
+		err = os.Unsetenv("TANZU_CONFIG")
+		assert.NoError(t, err)
+
+		err = os.Remove(tkgConfigFileNG.Name())
+		assert.NoError(t, err)
+		err = os.Unsetenv("TANZU_CONFIG_NEXT_GEN")
+		assert.NoError(t, err)
+	}
+	return cleanup
+}
+
+// TestGenerateKey tests the GenerateKey function.
+func TestEnsureMutualExclusiveCurrentContexts(t *testing.T) {
+	cleanup := setupMultiConcurrentContexts(t)
+
+	defer func() {
+		cleanup()
+	}()
```

Comment:
No need to wrap `cleanup()` in another `func`.  You can just do `defer cleanup()`

Reply:
Make sense as there is only one method for defer. Fixed.